### PR TITLE
cluster/gce: set KUBE_WATCHLIST_INCONSISTENCY_DETECTOR when requested by a CI job

### PR DESCRIPTION
**Pull request by p0lyn0mial**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
when the `KUBE_WATCHLIST_INCONSISTENCY_DETECTOR` environment variable was set then the reflector performs a data consistency check.

The consistency check is meant to be enforced only in the CI, not in production. The check ensures that data retrieved by the watch-list api call is exactly the same as data received by the standard list api call.

note that the new env var is set in the same places the `ENABLE_CACHE_MUTATION_DETECTOR` is set.

xref: https://github.com/kubernetes/enhancements/issues/3157
xfef: https://github.com/kubernetes/test-infra/pull/31082

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/3157-watch-list
```

### CODE REVIEW

File: cluster/gce/gci/configure-kubeapiserver.sh [hunk 1]
```diff
@@ -349,6 +349,12 @@ function start-kube-apiserver {
     fi
     container_env+="{\"name\": \"GODEBUG\", \"value\": \"${KUBE_APISERVER_GODEBUG}\"}"
   fi
+  if [[ -n "${ENABLE_KUBE_WATCHLIST_INCONSISTENCY_DETECTOR:-}" ]]; then
```
Comment (wojtek-t) [inline cluster/gce/gci/configure-kubeapiserver.sh:352]: nit: move it right below CACHE_MUTATION_DETECTOR

**Comment by wojtek-t** [comment]
/lgtm
/approve

/hold
in case you want to address the nit

**Comment by p0lyn0mial** [comment]
> /hold
in case you want to address the nit

addressed the comment, PTAL.

**Pull request merged**